app/api/dao/user.py

```diff
@@ -143,27 +146,25 @@ def get_user_by_username(username: str):
         return UserModel.find_by_username(username)
 
     @staticmethod
-    def list_users(user_id: int, search_query: str = "", is_verified = None):
+    def list_users(user_id: int, search_query: str = "", page: int = DEFAULT_PAGE, per_page: int = DEFAULT_USERS_PER_PAGE, is_verified = None):
         """ Retrieves a list of verified users with the specified ID.
         
         Arguments:
             user_id: The ID of the user to be listed.
             search_query: The search query for name of the users to be found.
             is_verified: Status of the user's verification; None when provided as an argument.
+            page: The page of users to be returned
+            per_page: The number of users to return per page
         
         Returns:
             A list of users matching conditions and the HTTP response code.
         
         """
 
-        users_list = UserModel.query.filter(UserModel.id != user_id).all()
+        users_list = UserModel.query.filter(UserModel.id != user_id, not is_verified or UserModel.is_email_verified, func.lower(UserModel.name).contains(search_query.lower())).paginate(page=page, per_page=per_page, error_out=False, max_per_page=UserDAO.DEFAULT_USERS_PER_PAGE).items
```

Comment:
this is amazing in terms of functionality 👏 you really tried to solve my comment @lmcicat , thank you for going above and behond :) I would just try to improve the identation for it to be a bit more readable. makes sense? @lmcicat 
I will test this, this weekend!

Reply:
hopefully what I changed made readability better!